packages/api/AlvTime.Persistence/Repositories/FlexhourStorage.cs

```diff
@@ -399,12 +397,161 @@ private decimal GetHoursAfterCompRate(List<OvertimeEntry> overtimeEntries, decim
+        {
+            if (hoursForPayoutCounter == requestedHoursForPayout)
+            {
+                break;
+            }
+
+            var indexForCurrentCompRate = indexForCompRateList[overtimeEntriesWithSalaryIndex].indexOfComprate;
+            var overtimeEntriesWithSalaryOrderedByDate = overtimeEntriesWithSalaryGroupedByCompRate[indexForCurrentCompRate].OrderBy(x=>x.OvertimeEntry.Date).ToList();
+            for (int overtimeEntryIndex = 0; overtimeEntryIndex < overtimeEntriesWithSalaryOrderedByDate.Count; overtimeEntryIndex++)
+            {
+                if (hoursForPayoutCounter == requestedHoursForPayout)
+                {
+                    break;
+                }
+
+                if (overtimeEntriesWithSalaryOrderedByDate[overtimeEntryIndex].OvertimeEntry.Hours + hoursForPayoutCounter > requestedHoursForPayout)
+                {
+                    overtimeEntriesForPayoutCalculation.Add(new OvertimeEntryWithSalary(
+                        overtimeEntriesWithSalaryOrderedByDate[overtimeEntryIndex].SalaryPrHour,
+                        new OvertimeEntry
+                        {
+                            CompensationRate = overtimeEntriesWithSalaryOrderedByDate[overtimeEntryIndex].OvertimeEntry.CompensationRate,
+                            Date = overtimeEntriesWithSalaryOrderedByDate[overtimeEntryIndex].OvertimeEntry.Date,
+                            Hours = requestedHoursForPayout - hoursForPayoutCounter,
+                            Salary = overtimeEntriesWithSalaryOrderedByDate[overtimeEntryIndex].OvertimeEntry.Salary,
+                            TaskId = overtimeEntriesWithSalaryOrderedByDate[overtimeEntryIndex].OvertimeEntry.TaskId
+                        }));
+
+                    hoursForPayoutCounter += (requestedHoursForPayout - hoursForPayoutCounter);
+                }
+                else if (overtimeEntriesWithSalaryOrderedByDate[overtimeEntryIndex].OvertimeEntry.Hours + hoursForPayoutCounter <= requestedHoursForPayout)
+                {
+                    hoursForPayoutCounter += overtimeEntriesWithSalaryOrderedByDate[overtimeEntryIndex].OvertimeEntry.Hours;
+                    overtimeEntriesForPayoutCalculation.Add(overtimeEntriesWithSalaryOrderedByDate[overtimeEntryIndex]);
+                }
+            }
+        }
+        return CalculatePayout(overtimeEntriesForPayoutCalculation);
+    }
+    
+    private decimal CalculatePayout(List<OvertimeEntryWithSalary> overtimeEntriesWithSalary)
+    {
+        return overtimeEntriesWithSalary.Sum(overtimeEntry => overtimeEntry.OvertimeEntry.CompensationRate * overtimeEntry.OvertimeEntry.Hours * overtimeEntry.SalaryPrHour);
+    }
+
+    public decimal RegisterOvertimePayout(List<OvertimeEntry> overtimeEntries, int userId, GenericHourEntry requestedPayout, int paidOvertimeId)
+    {
+        var salaryData = _salaryService.GetEmployeeSalaryData(userId).OrderBy(x => x.FromDate).ToList();

```

Comment:
Vi kan sikkert refaktorere dette senere, men det ser litt funky ut at flyten går fra service -> repository -> service -> repository. Optimalt burde det kun vært èn gang innom hvert sted. Men bare la det være sånn for nå, hele FlexhourStorage er jo bare megaspagetti så det må gjøres noe med på en litt større skala 😄

Reply:
Ja, jeg er enig i at dette ikke er optimalt. Det ble litt fusjon av pre-onion architectur og onion architecture. Tenker at jeg kan refakturere litt når det kommer